### PR TITLE
Send To Manager Review Now Making Use of CompanyReviewers Table

### PR DESCRIPTION
**Pull request by carlosmorales125**

### CODE REVIEW

File: src/Aquifer.Data/Entities/UserEntity.cs [hunk 1]
```diff
@@ -29,6 +28,8 @@ public class UserEntity : IHasUpdatedTimestamp
     [InverseProperty(nameof(CompanyEntity.DefaultReviewerUser))]
     public ICollection<CompanyEntity> CompaniesAsDefaultReviewer { get; set; } = [];
 
+    public ICollection<CompanyReviewerEntity> CompanyReviewers { get; set; } = [];
```
Comment (MrGrinst) [inline src/Aquifer.Data/Entities/UserEntity.cs:31]: Do we need this here? Doesn't look like it's being referenced anywhere.

File: src/Aquifer.Data/Entities/CompanyReviewerEntity.cs [hunk 2]
```diff
@@ -12,8 +13,13 @@ public class CompanyReviewerEntity
 
     [SqlDefaultValue("getutcdate()")]
     public DateTime Created { get; set; } = DateTime.UtcNow;
-
+    
+    [ForeignKey(nameof(UserId))]
```
Comment (jwinston-bn) [inline src/Aquifer.Data/Entities/CompanyReviewerEntity.cs:17]: Don't think we need these attributes.

**Review by MrGrinst — APPROVED** [review]

**Review by MrGrinst — CHANGES_REQUESTED** [review]
Could you also remove the DefaultReviewerUser stuff from CompanyEntity? We don't want to do the migration to remove the column in this PR, but it would be nice to deploy the change that removes code-level references alongside this.

**Comment by jwinston-bn** [comment]
> Could you also remove the DefaultReviewerUser stuff from CompanyEntity? We don't want to do the migration to remove the column in this PR, but it would be nice to deploy the change that removes code-level references alongside this.

Doesn't that impact the foreign key relationship though, and by removing it, wouldn't that generate a change in the migration file?

**Comment by MrGrinst** [comment]
> > Could you also remove the DefaultReviewerUser stuff from CompanyEntity? We don't want to do the migration to remove the column in this PR, but it would be nice to deploy the change that removes code-level references alongside this.
> 
> Doesn't that impact the foreign key relationship though, and by removing it, wouldn't that generate a change in the migration file?

It would only affect the migrations if we ran the migration add command. But to avoid the 500s we got last time we removed a column I think we need to remove the code-level reference in one PR, then do the migration in a separate PR.

**Comment by MrGrinst** [comment]
> Yeah, I'm struggling to think of what the ordering should be here. It doesn't feel right for an entity to be in prod that doesn't align with what's in the database. Is your suggestion to also remove `DefaultReviewerUserId` from CompanyEntity now as well?

Yeah we would remove both of them, I should've been clearer but that was my suggestion to remove `DefaultReviewerUserId` and `DefaultReviewerUser`. And the entity will only be in prod for a very short duration before we would deploy the migration to remove the column.

**Comment by jwinston-bn** [comment]
All right. I think that's fine. So my suggestion would be to remove it now, and then have a follow-up PR ready to go up to prod with the migration to deploy immediately after the first goes up.

**Review by jwinston-bn — APPROVED** [review]

**Review by MrGrinst — APPROVED** [review]

**Pull request merged**